### PR TITLE
Fix 'PromiseState.refresh should return a new PromiseState'

### LINKED ISSUE #149: PromiseState.refresh should return a new PromiseState
PromiseState.refresh should return a new PromiseState because this way a `shallowEqual` comparison of the props in the wrapped component erroneously returns `true` even though the PromiseState changed

### PR DESCRIPTION
**Pull request by ryanbrainard**
Fixes #149 

cc: @nfcampos

### CODE REVIEW

File: src/PromiseState.js [hunk 1]
```diff
@@ -11,9 +11,17 @@ export default class PromiseState {
   // creates as PromiseState that is refreshing
   // can be called without a previous PromiseState and will be both pending and refreshing
   static refresh(previous, meta) {
```
Comment (ryanbrainard) [inline src/PromiseState.js]: I didn't address it here, but was noticing how it's a little weird that `meta` here is only used if `previous` is undefined; although, it does act like `resolve(value, meta)` though. I believe I did that this to simplify some of the logic on the refetch size, but it makes this API a little ambiguous. Don't think it needs to be addressed now, but just calling it out for the future.

Comment (nfcampos) [inline src/PromiseState.js]: Yeah that's a little weird

**Comment by nfcampos** [comment]
This looks great! 👍

**Comment by ryanbrainard** [comment]
Released: https://github.com/heroku/react-refetch/releases/tag/v1.0.0-beta.10

**Comment by nfcampos** [comment]
Thanks!

**Pull request merged**